components/net/fetch/methods.rs

```diff
@@ -498,3 +530,94 @@ fn is_null_body_status(status: &Option<StatusCode>) -> bool {
+            let type_ = request.type_;
+
+            /// This is a tricky one: it is not straightforward.
+            /// The current practice is to use file sniffing
+            /// to determine the actual font file type.
+            ///
+            /// There is a [proposal](https://www.w3.org/TR/WOFF2/#IMT)
+            /// to add a top level "font/" MIME type.
+            ///
+            /// [These](https://docs.google.com/document/d/1Tsju6EOP4LqJ1RcFJRpqxryggbWZYbq5jvsIhVYNoyU/)
+            /// are the most commonly used MIME types for fonts in practice.
+            ///
+            /// TODO determine how to handle font file acceptance with MIME types
+            #[inline]
+            fn is_font_mime_type(mime: &Mime) -> bool {
+                // Declare that media MIME types are invalid for fonts
+                match mime.0 {
+                    TopLevel::Audio | TopLevel::Image | TopLevel::Video => false,
+                    _ => true
+                }
+            }
+
+            /// https://html.spec.whatwg.org/multipage/#scriptingLanguages
+            #[inline]
+            fn is_javascript_mime_type(mime_type: &Mime) -> bool {
+                // TODO: try to replace with const
+                let javascript_mime_types: [Mime; 16] = [
+                    mime!(Application / ("ecmascript")),
+                    mime!(Application / ("javascript")),
+                    mime!(Application / ("x-ecmascript")),
+                    mime!(Application / ("x-javascript")),
+                    mime!(Text / ("ecmascript")),
+                    mime!(Text / ("javascript")),
+                    mime!(Text / ("javascript1.0")),
+                    mime!(Text / ("javascript1.1")),
+                    mime!(Text / ("javascript1.2")),
+                    mime!(Text / ("javascript1.3")),
+                    mime!(Text / ("javascript1.4")),
+                    mime!(Text / ("javascript1.5")),
+                    mime!(Text / ("jscript")),
+                    mime!(Text / ("livescript")),
+                    mime!(Text / ("x-ecmascript")),
+                    mime!(Text / ("x-javascript")),
+                ];
+
+                javascript_mime_types.contains(mime_type)
+            }
+
+            // TODO: try to replace with const
+            let text_css: Mime = mime!(Text / Css);
+            let text_vtt: Mime = mime!(Text / ("vtt"));
+            // Assumes str::starts_with is equivalent to mime::TopLevel
+            return match type_ {
+                // Step 7
+                Type::Image => mime_type.0 != TopLevel::Image,
+                // Step 8
+                Type::Font => !is_font_mime_type(&mime_type),
+                // Step 9
+                Type::Script => !is_javascript_mime_type(&mime_type),
+                // Step 10
+                Type::Style => mime_type != &text_css,
+                // Step 11
+                Type::Track => mime_type != &text_vtt,
+                // Step 12
```

Comment:
Where are these steps coming from? They don't appear in https://fetch.spec.whatwg.org/#should-response-to-request-be-blocked-due-to-nosniff? at all.